src/containers/pages/AssigmentMapWrapper/index.tsx

```diff
@@ -1,49 +1,97 @@
-import { CountriesAdmin0 } from '../../../../src/configs/settings';
+import { useHistory } from 'react-router';
+import { ActionCreator, Store } from 'redux';
+import { CountriesAdmin0, PlanDefinition } from '../../../../src/configs/settings';
 import { MemoizedGisidaLite } from '../../../components/GisidaLite';
 import Loading from '../../../components/page/Loading';
 import { getJurisdictions } from '../../../components/TreeWalker/helpers';
 import { MAP_LOAD_ERROR } from '../../../configs/lang';
 import { displayError } from '../../../helpers/errors';
 import { OpenSRPService } from '../../../services/opensrp';
-import { Filters, getCurrentChildren } from '../../../store/ducks/opensrp/hierarchies';
-import { TreeNode } from '../../../store/ducks/opensrp/hierarchies/types';
-import { nodeIsSelected } from '../../../store/ducks/opensrp/hierarchies/utils';
+import {
+  autoSelectNodes,
+  AutoSelectNodesAction,
+  deselectNode,
+  DeselectNodeAction,
+  Filters,
+  getCurrentChildren,
+  getCurrentParentNode,
+  getMetaData,
+  selectNode,
+  SelectNodeAction,
+} from '../../../store/ducks/opensrp/hierarchies';
+import { Meta, TreeNode } from '../../../store/ducks/opensrp/hierarchies/types';
 import jurisdictionReducer, {
   fetchJurisdictions,
   Filters as JurisdictionGeomFilters,
   getJurisdictionsById,
   getJurisdictionsFC,
   reducerName as jurisdictionReducerName,
 } from '../../../store/ducks/opensrp/jurisdictions';
-import { buildStructureLayers } from '../FocusInvestigation/map/active/helpers/utils';
+import {
+  buildMouseMoveHandler,
+  buildStructureLayers,
+} from '../FocusInvestigation/map/active/helpers/utils';
+import { onJurisdictionClick } from './helpers/utils';
+
 reducerRegistry.register(jurisdictionReducerName, jurisdictionReducer);
 
 /** props for Plan jurisdiction and team assignment higher order component */
 export interface AssignmentMapWrapperProps {
+  autoSelectionFlow: boolean;
+  plan: PlanDefinition;
   rootJurisdictionId: string;
-  currentParentId: string | undefined;
+  currentParentId?: string;
+  jurisdictionsChunkSize: number;
+  getJurisdictionsMetadata: Dictionary<Dictionary<Dictionary<Meta>>>;
   currentChildren: TreeNode[];
   serviceClass: typeof OpenSRPService;
   fetchJurisdictionsActionCreator: typeof fetchJurisdictions;
   getJurisdictionsFeatures: FeatureCollection;
+  autoSelectNodesActionCreator: ActionCreator<AutoSelectNodesAction>;
+  selectNodeCreator: ActionCreator<SelectNodeAction>;
+  deselectNodeCreator: ActionCreator<DeselectNodeAction>;
+  currentParentNode?: TreeNode;
 }
 
-const defaultProps = {
+/** default value for feature Collection */
+const defaultFeatureCollection: FeatureCollection = {
+  features: [],
+  type: 'FeatureCollection',
+};
+
+const defaultProps: AssignmentMapWrapperProps = {
+  autoSelectNodesActionCreator: autoSelectNodes,
+  autoSelectionFlow: false,
   currentChildren: [],
   currentParentId: undefined,
+  currentParentNode: undefined,
```

Comment:
I do not think it is necessary to declare `currentParentNode` as undefined, as that is already catered for in the interface.